From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: suvojeet-sengupta <suvojitsengupta21@gmail.com>
Date: Mon, 19 Oct 2026 15:40:56 +0000
Subject: [PATCH] SettingsProvider: Let devices set the default minimum refresh
 rate

Load Settings.System.MIN_REFRESH_RATE from def_min_refresh_rate on first boot, so a device can ship with its peak refresh rate forced (adaptive off) and still let the user turn adaptive back on.

Change-Id: I5c27f27677d12c6ee89a5b74f301325c9bd4a14b
---
 packages/SettingsProvider/res/values/defaults.xml              | 3 +++
 .../src/com/android/providers/settings/DatabaseHelper.java     | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/packages/SettingsProvider/res/values/defaults.xml b/packages/SettingsProvider/res/values/defaults.xml
index f6d72070e68d..0ab4e1a4bbff 100644
--- a/packages/SettingsProvider/res/values/defaults.xml
+++ b/packages/SettingsProvider/res/values/defaults.xml
@@ -134,6 +134,9 @@
     <!-- Default for Settings.System.POINTER_SPEED -->
     <integer name="def_pointer_speed">0</integer>
 
+    <!-- Default for Settings.System.MIN_REFRESH_RATE, 0 lets the display drop to lower rates -->
+    <integer name="def_min_refresh_rate">0</integer>
+
     <!-- Default for DTMF tones enabled -->
     <bool name="def_dtmf_tones_enabled">true</bool>
     <!-- Default for UI touch sounds enabled -->
diff --git a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
index 5a4bd6bdd91a..b043aa93918f 100644
--- a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
+++ b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
@@ -2135,6 +2135,9 @@ class DatabaseHelper extends SQLiteOpenHelper {
             loadIntegerSetting(stmt, Settings.System.POINTER_SPEED,
                     R.integer.def_pointer_speed);
 
+            loadIntegerSetting(stmt, Settings.System.MIN_REFRESH_RATE,
+                    R.integer.def_min_refresh_rate);
+
             /*
              * IMPORTANT: Do not add any more upgrade steps here as the global,
              * secure, and system settings are no longer stored in a database
-- 
2.43.0

